userland/Makefile

```diff
@@ -31,25 +32,67 @@ CPPFLAGS += \
 	    -mpic-register=r9\
 	    -mno-pic-data-is-text-relative
 
+$(BUILDDIR)/%.o: %.c | $(BUILDDIR)
+	$(TRACE_DEP)
+	$(Q)$(CC) $(CFLAGS) $(CPPFLAGS) -MF"$(@:.o=.d)" -MG -MM -MP -MT"$(@:.o=.d)@" -MT"$@" "$<"
```

Comment:
Any chance you could add a comment here to explain what this is doing? Why are there all of these arguments here that aren't part of `CFLAGS` or `CPPFLAGS`?